src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -206,8 +203,29 @@ private boolean endNodeHasAmenityTypeToExclude(final AtlasObject object)
         final Edge edge = (Edge) object;
         final Node endNode = edge.end();
 
-        return Validators.isOfType(endNode, AmenityTag.class, AmenityTag.PARKING,
-                AmenityTag.PARKING_SPACE, AmenityTag.MOTORCYCLE_PARKING,
-                AmenityTag.PARKING_ENTRANCE);
+        return Validators.isOfType(endNode, AmenityTag.class, amenityValuesToExclude);
+    }
+
+    /**
+     * This function checks an edge to determine whether it has certain characteristics that signify
+     * to us that we do not want to keep examining this component of the network.
+     *
+     * @param edge
+     *            An Edge we're examining
+     * @return {@code true} if the edge is already flagged, has an amenity type we want to exclude,
+     *         or ends in a boundary node {@code false} otherwise
+     */
+    private boolean shouldQuitNow(final Edge edge)
```

Comment:
Can we come up with a better name? 😆

Reply:
Struggling a little bit with the naming here -- I tried another name but am definitely open to suggestions!